src/tools/crawler/metrics.rs

```diff
@@ -48,7 +52,7 @@ pub fn new_network_summary(crawler: &Crawler, graph: &Graph<SocketAddr>) -> Netw
     let mut protocol_versions = HashMap::with_capacity(num_known_nodes);
     let mut user_agents = HashMap::with_capacity(num_known_nodes);
 
-    for (_, node) in nodes {
+    for (_, node) in nodes.clone() {
```

Comment:
can we use `iter()` here? 🕵️‍♀️

Reply:
Yes, we can. Then `user_agent` needs to be cloned.
Done.